product_variant_configurator/models/product_configurator.py

```diff
@@ -185,8 +185,15 @@ def unlink(self):
             attributes.unlink()
         return result
 
+    @api.multi
+    def check_configuration_validity(self):
+        for rec in self:
+            if any(not x.value_id for x in rec.product_attribute_ids):
+                raise exceptions.ValidationError(
+                    _("You have to fill all the attributes values."))
+
     @api.model
-    def check_configuration_validity(self, vals):
+    def check_configuration_validity_from_vals(self, vals):
```

Comment:
You have also the methods `_convert_to_write` and `_convert_to_read` that can switch from dictionary to record and vice versa to handle things in a more homogeneous way and avoid duplicated code.